src/Diagnostics.RuntimeHost/Services/DevOpsClient/DevOpsClient.cs

```diff
@@ -0,0 +1,239 @@
+﻿using Diagnostics.RuntimeHost.Utilities;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace Diagnostics.RuntimeHost.Services.DevOpsClient
+{
+    public interface IDevOpsClient : IDisposable, IRepoClient
+    {
+        string AccessToken { get; }
+        string Organization { get; }
+        string RepositoryID { get; }
+        string Project { get; }
+    }
+
+    public class DevOpsResponse
+    {
+        public object result;
+        public HttpStatusCode responseCode;
+    }
+
+    class DevOpsClient : IDevOpsClient
+    {
+        private static string _accessToken;
+        private static string _organization;
+        private static string _repoID;
+        private static string _project;
+        private static IConfiguration _config;
+        private static VssCredentials credentials;
+        private static VssConnection connection;
+        private GitHttpClient gitClient;
+
+        public DevOpsClient(IConfiguration config)
+        {
+            _config = config;
+            LoadConfigurations(config);
+            credentials = (VssCredentials)(FederatedCredential)new VssBasicCredential("pat", _accessToken);
+            connection = new VssConnection(new Uri($"https://dev.azure.com/{_organization}"), credentials);
```

Comment:
`dev.azure.com` may not be the prefix every time. In some cases it changes based on a team's set up. Eg; it can be msazure.visualstudio.com/<org>/<repo>. I recommend storing the entire domain as config and pull from that. In the future we can save each partner's URL as a config property.

Reply:
I tried accessing a repo on `msazure.visualstudio.com` with the API using `dev.azure.com` and was successful. I believe this format will work even if the domain is different.

Reply:
Thats great. Then we can stick to this